core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -61,19 +86,51 @@
     try {
       return callable.call();
     } catch (AlluxioException e) {
-      LOG.debug("Internal Alluxio error when running rpc", e);
+      LOG.debug("Internal Error: {}", callable, e);
+      throw e.toThrift();
+    } catch (IOException e) {
+      LOG.debug("I/O Error: {}", callable, e);
+      throw new ThriftIOException(e.getMessage());
+    } catch (Exception e) {
+      LOG.error("Unexpected Error: {}", callable, e);
+      throw new UnexpectedAlluxioException(e).toThrift();
+    }
+  }
+
+  /**
+   * Calls the given {@link RpcCallableThrowsIOException} and handles any exceptions thrown. This
+   * method also logs enter and exit when debug level logging is enabled.
+   *
+   * @param logger the logger to use for this call
+   * @param callable the callable to call
+   * @param <T> the return type of the callable
+   * @return the return value from calling the callable
+   * @throws AlluxioTException if the callable throws an Alluxio or runtime exception
+   * @throws ThriftIOException if the callable throws an IOException
+   */
+  public static <T> T call(Logger logger, RpcCallableThrowsIOException<T> callable)
```

Comment:
I think I will take Bin's suggestion and make two separate methods.

The reason why I put `logger` as the first parameter is to make the method call sites look cleaner in terms of style. If you think having `logger` as the first argument is counterintuitive even if this method is not overloaded, I'm fine with changing it.

The only reason why `logger` needs to be passed in the first place is to have the log message print the actual class ie. `FileSystemMasterClientServiceHandler` as opposed to `RPCUtils`. I think its valuable but I just want to point out why I did so in case you guys think it is not worth the complexity in `RPCUtils`.

The reason why I cannot just pass `LOG` as the default argument is because we do not want enter/exit debug level logging for certain RPCs like `sessionHeartbeat` since those will be far too frequent.